icu4j/main/tests/core/src/com/ibm/icu/dev/test/charsetdet/TestCharsetDetector.java

```diff
@@ -1249,5 +1250,55 @@ public void run() {
         }
     }
 
+    @Test
+    public void plainAsciiLeadsToAsciiBeingFirst() throws Exception {
+        // Content: "@{,xxx-yyy-!!!}"
+        // Non-English text because otherwise, it is detected as ISO-8859-1
+        byte[] input = new byte[]{0x40, 0x7b, 0x2c, 0x78, 0x78, 0x78, 0x2d, 0x79, 0x79, 0x79, 0x2d, 0x21, 0x21, 0x21, 0x7d};
+
+        CharsetDetector charsetDetector = new CharsetDetector();
+        charsetDetector.setText(input);
+        CharsetMatch[] matches = charsetDetector.detectAll();
+
+        // "just" check that ASCII is first
+        Assert.assertEquals("ASCII", matches[0].getName());
+        Assert.assertEquals(35, matches[0].getConfidence());
+    }
+
+    @Test
+    public void utf8LeadsToAsciiNotBeingContained() throws Exception {
+        // Content: "This is a UTF-8 string! 🎉"
+        byte[] input = new byte[]{0x54, 0x68, 0x69, 0x73, 0x69, 0x73, 0x61, 0x55, 0x54, 0x46, 0x2d, 0x38, 0x73, 0x74, 0x72, 0x69, 0x6e, 0x67, 0x21, (byte) 0xf0, (byte) 0x9f, (byte) 0x8e, (byte) 0x89};
```

Comment:
optional / more readable:
```suggestion
        byte[] input = "This is a UTF-8 string! 🎉".getBytes(StandardCharsets.UTF_8);
```